polaris-react/src/components/Filters/components/SearchField/SearchField.tsx

```diff
@@ -57,52 +50,29 @@ export function SearchField({
   }
 
   return (
-    <div className={styles.SearchField}>
-      <label className={styles.Label} htmlFor={id}>
-        {placeholder}
-      </label>
-      <input
-        id={id}
-        ref={inputRef}
-        className={classNames(
-          styles.Input,
-          focused && styles.focused,
-          borderlessQueryField && styles.borderless,
-        )}
-        value={value}
-        onChange={(event) => handleChange(event?.currentTarget.value ?? value)}
-        onFocus={onFocus}
-        onBlur={onBlur}
-        autoComplete="off"
-        placeholder={placeholder}
-        disabled={disabled}
-      />
-      {loading || value !== '' ? (
-        <div className={styles.Suffix}>
-          <InlineStack gap="200">
-            {loading ? (
-              <div className={styles.Spinner}>
-                <Spinner size="small" />
-              </div>
-            ) : null}
-            {value !== '' ? (
-              <UnstyledButton
-                className={classNames(
-                  styles.ClearButton,
-                  focused && styles['ClearButton-focused'],
-                )}
-                onClick={() => handleClear()}
-                disabled={disabled}
-              >
-                <Text as="span" visuallyHidden>
```

Comment:
the clear button gets ported over to `TextField` via the `clearButton` prop. This code is nearly identical to what is already in [`TextField`](https://github.com/Shopify/polaris/blob/main/polaris-react/src/components/TextField/TextField.tsx#L357-L372). There are also bugs with this current custom implementation